src/v0/sources/braze/transform.js

```diff
@@ -40,7 +40,9 @@ const processEvent = (event) => {
 
     // set timestamp for the event
     if (event.time) {
-      message.setProperty('timestamp', formatTimeStamp(event.time, 'YYYY-MM-DD[T]HH:mm:ss.SSS[Z]'));
+      // Refer: https://www.braze.com/docs/user_guide/data_and_analytics/braze_currents/event_glossary/message_engagement_events
+      const timestamp = parseInt(event.time, 10) * 1000;
```

Comment:
I was trying to do similar thing. WDYT of the below changes instead of this
File: `v0/util/index.js`
```javascript
// Important !@!
// Generic timestamp formatter
const formatTimeStamp = (dateStr, format) => {
  let date = new Date(dateStr);
  if (_.isNumber(dateStr)) {
    date = new Date(dateStr * 1000)
  }
  // moment format is passed. format accordingly
  if (format) {
    return moment.utc(date).format(format);
  }

  // return default format
  return date.getTime();
};
```

Reply:
This would any such issues for all who are using `formatTimeStamp` function. Let me know your thoughts @koladilip

Reply:
but in common util we can't assume that timestamp always comes in seconds right? in case of braze we know for sure that they will be sending seconds

Reply:
True. The condition `_.isNumber` handles it for us as well, if a number(like epoch) is sent, it would be correctly formatted. That was my thinking

Reply:
Feel free to ignore this comment

Reply:
timestamp could in seconds or mills seconds right, even though it is number, I wouldn't assume that it will be always seconds in common util.